src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -259,12 +259,19 @@ bool VerifyCrlSignature(X509_CRL* crl, X509* issuer) {
   if (ikey == nullptr) {
     // Can't verify signature because we couldn't get the pubkey, fail the
     // check.
+    gpr_log(GPR_DEBUG, "Could not public key from certificate.");
     EVP_PKEY_free(ikey);
     return false;
   }
-  bool ret = X509_CRL_verify(crl, ikey) == 1;
+  int ret = X509_CRL_verify(crl, ikey);
+  if (ret < 0) {
+    gpr_log(GPR_DEBUG,
+            "There was an unexpected problem checking the CRL signature.");
+  } else if (ret == 0) {
+    gpr_log(GPR_DEBUG, "CRL could not be verified.");
+  }
   EVP_PKEY_free(ikey);
-  return ret;
+  return ret == 1;
```

Comment:
Do we need a unit test that detects this functional change? Said differently, do we have any test CRL that would trigger `X509_CRL_verify` to return -1?

Reply:
It's not a behavior change, previously the line was `bool ret = X509_CRL_verify(crl, ikey) == 1;`

Now we just temporarily look at the `int` because a `0` return means verification failed, but `-1` indicates that something happened that it couldn't verify at all.

Reply:
Whoops, I misread! Thanks for clarifying!